spinalcordtoolbox/types.py

```diff
@@ -382,10 +381,12 @@ def compute_vertebral_distribution(self, discs_levels, label_reference='C1'):
                 if index_last_label is None or index_label > index_last_label:
                     index_last_label = index_label
 
-        if index_first_label is not None:
-            self.first_label = self.list_labels[index_first_label]
-        if index_last_label is not None:
-            self.last_label = self.list_labels[index_last_label]
+        # Assign first/last labels once we guarantee that there even *are* valid labels
+        if index_last_label is None or index_last_label is None:
+            raise ValueError(f"None of the provided disc levels {[l[3] for l in discs_levels]} are present in the"
+                             f"list of expected disc levels: {list(self.list_labels)}.")
+        self.first_label = self.list_labels[index_first_label]
+        self.last_label = self.list_labels[index_last_label]
```

Comment:
This change was made so that we could reliably access the value of `self.first_label` later in the function.

(Before, it was unclear whether `self.first_label` would even be assigned.)